scripts/lib/CIME/case_submit.py

```diff
@@ -61,6 +63,10 @@ def _submit(case, job=None, no_batch=False, prereq=None, resubmit=False,
         if case.get_value("TEST"):
             case.set_value("IS_FIRST_RUN", False)
 
+    if env_batch.get_batch_system_type() != "none":
+        # May need to regen batch files if user made walltime changes
```

Comment:
walltime is misleading here - the changes could be anything in env_batch: - job_queue, wallclock time, handling of script error and output files - really anything that could be changed in that file.